code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -111,13 +111,21 @@
 		update_inv_gloves()
 	else if(I == glasses)
 		glasses = null
+		var/obj/item/clothing/glasses/G = I
+		if(G.tint)
+			update_tint()
+		if(G.prescription)
+			clear_fullscreen("nearsighted")

```

Comment:
Welding overlays cover a different fullscreen index than `nearsighted` does - removing the `nearsighted` fullscreen won't touch a blurred screen from being inside an object, having the `eye_blur` status set, or welding equipment.